ecs-cli/modules/cli/local/network/teardown.go

```diff
@@ -0,0 +1,147 @@
+// Copyright 2015-2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//	http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package network
+
+import (
+	"strings"
+	"time"
+
+	"github.com/docker/docker/api/types"
+	"github.com/sirupsen/logrus"
+	"golang.org/x/net/context"
+)
+
+// LocalEndpointsStopper groups the Docker NetworkInspect, ContainerStop, ContainerRemove, and NetworkRemove functions.
+//
+// These functions can be used together to remove a network once unwanted containers are stopped.
+type LocalEndpointsStopper interface {
+	networkInspector
+	containerStopper
+	containerRemover
+	networkRemover
+}
+
+type networkInspector interface {
+	NetworkInspect(ctx context.Context, networkID string, options types.NetworkInspectOptions) (types.NetworkResource, error)
+}
+
+type containerStopper interface {
+	ContainerStop(ctx context.Context, containerID string, timeout *time.Duration) error
+}
+
+type containerRemover interface {
+	ContainerRemove(ctx context.Context, containerID string, options types.ContainerRemoveOptions) error
+}
+
+type networkRemover interface {
+	NetworkRemove(ctx context.Context, networkID string) error
+}
+
+// Teardown removes both the Local Endpoints container and the Local network created by Setup.
+// If there are other containers running in the network besides the endpoints container, this function does nothing.
+//
+// If there is any unexpected errors, we exit the program with a fatal log.
+func Teardown(dockerClient LocalEndpointsStopper) {
+	if hasRunningTasksInNetwork(dockerClient) {
+		return
+	}
+	logrus.Infof("The network %s has no more running tasks, stopping the endpoints containers...", EcsLocalNetworkName)
+
+	stopEndpointsContainer(dockerClient)
+	removeEndpointsContainer(dockerClient)
+	removeLocalNetwork(dockerClient)
+}
+
+// hasRunningTasksInNetwork returns true if there are other containers besides the
+// endpoints container running in the local network, false otherwise.
+func hasRunningTasksInNetwork(d networkInspector) bool {
+	ctx, cancel := context.WithTimeout(context.Background(), dockerTimeout)
+	defer cancel()
+
+	resp, err := d.NetworkInspect(ctx, EcsLocalNetworkName, types.NetworkInspectOptions{})
+	if err != nil {
+		logrus.Fatalf("Failed to inspect network %s due to %v", EcsLocalNetworkName, err)
+	}
+
+	if len(resp.Containers) > 1 {
+		// Has other containers running in the network
+		return true
```

Comment:
I'd suggest adding a log statement here so that users have more visibility in the case where teardown is being skipped because they had launched other containers in the network

Reply:
Done! Added `"%d other task(s) running locally, skipping network removal."`